packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -389,7 +403,14 @@ class Auth {
         if (refreshToken) {
             try {
                 return await this.queueRequest(
-                    () => helpers.refreshAccessToken(this.client, {refreshToken}),
+                    () =>
+                        helpers.refreshAccessToken(
+                            this.client,
+                            {refreshToken},
+                            {
+                                clientSecret: this.clientSecret ? this.clientSecret : undefined
```

Comment:
Side note: I wonder why we chose to user an empty string as the default value for clientSecret and fetchedToken etc. It's good that you copied this for consistency but it prevents us from simply doing 

```
clientSecret: this.clientSecret
```

It kinda shows that we are handling undefined values in 2 different ways. Which isn't great. 

This is something we can fix later.

Reply:
Good point, we can definitely pass this.clientSecret without the ternary condition here. The ternary was outdated, I used it for an old approach I had before.